src/Messaging/utils.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { createDestination, parseURIPattern, validateEventContext, validateEventType } from "./utils";
+import { IllegalStateError, IllegalArgumentError } from "../Errors";
+import Carbon from "../Carbon";
+
+describe( "validateEventContext", ():void => {
```

Comment:
Why is this not using our conventions for creating tests/documentation?

Reply:
This methods are just for internal use, and are not been exported in the `Messaging` module, thus I think it makes sense to leave them undocumented for the API Reference.

Reply:
If that's the case then I think we should move them to `Messaging.Utils` and expose them even if we are only using them internally. There are a lot of functions that we are exposing that we don't expect users to use but they are being used by other services.